### PR TITLE
Add search to usersManagement

### LINKED ISSUE #6674: Filtering and sorting of users list
# Description

## User Stories

* > As an admin I want to identify a secific user in the users list, so that I can later modify their data.

## Value
- user list filtering & sorting

## Acceptance Criteria
- Columns can be sorted (exept actions column)
- Filter
  - Placeholder text and aria-label: "Search users" 
  - userlist gets filtered with every keystroke
  - filtered columns are: Username, Email, Name. ~~Status~~
  - matching searchphrase gets highlighted in every result in the respective fields (bkulmann hat codeexamples for this)

## Definition of ready
[ ] everybody needs to understand the value written in the user story
[ ] acceptance criteria has to be defined
[ ] all dependencies of the user story need to be identified
[ ] feature should be seen from an end user perspective
[ ] user story has to be estimated
[ ] story points need to be less then 20

## Definition of done
- Functional requirements
[ ] functionality described in the user story works
[ ] acceptance criteria are fulfilled
- Quality
[ ] codre review happened
[ ] CI is green
[ ] critical code received unit tests by the developer
[ ] automated tests passed (if automated tests are not available, this test needs to be created and passed
- Non-functional requirements
[ ] no sonar cloud issues

### PR DESCRIPTION
**Pull request by fschade**
handover @JanAckermann 

## Description
image

## Related Issue

- Fixes #6674

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: packages/web-app-user-management/src/components/Users/UsersList.vue
Comment (kulmann) [inline packages/web-app-user-management/src/components/Users/UsersList.vue:157]: Would also make sense to filter by role names if possible

File: packages/web-app-user-management/src/components/Users/UsersList.vue
Comment (kulmann) [inline packages/web-app-user-management/src/components/Users/UsersList.vue:150]:
```suggestion
      if (!(searchTerm || '').trim()) {
```

File: packages/web-app-user-management/src/search/filter/index.ts
Comment (kulmann) [inline packages/web-app-user-management/src/search/filter/index.ts:32]: This doesn't work in transifex. I'm afraid you have to go the repetitive way here with:
```js
switch(kind(this.router.currentRoute)) {
  case "users": return $gettext("Search users ↵")
  case "groups": return $gettext("Search groups ↵")
}
```
otherwise those strings will never land in transifex. additionally extraction from .ts files doesn't work with our current setup, so it's probably broken anyway 🙈

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/web-app-user-management/src/components/Users/UsersList.vue
Comment (JanAckermann) [reply packages/web-app-user-management/src/components/Users/UsersList.vue:157]: Will be done later

**Review by kulmann — CHANGES_REQUESTED** [review]
one weird effect: mark.js highlights all occurrences of the search term, while the filtering doesn't work that way. This leads to filtering by `admin`, not seeing all users that have the admin role (only those who have admin in their username, name or email) but still getting the role `Admin` highlighted. Another indicator that we should refactor the role into the user, but of course for a separate PR and not something that blocks this PR. Could you open a bug ticket please?

Filtering and highlighting works really well.

I see a contrast issue though: in dark mode the text is barely readable. Please open a bug ticket for that as well.

Changelog item is missing (for that I have to `request changes` 😅 will approve once you added it)

**Review by kulmann — APPROVED** [review]
😍

**Pull request merged**
